ledger/store/trackerdb/sqlitedriver/kvsIter.go

```diff
@@ -62,29 +63,73 @@ func (iter *kvsIter) Close() {
 
 // tableIterator is used to dump onlineaccounts and onlineroundparams tables for catchpoints.
 type tableIterator[T any] struct {
-	rows *sql.Rows
-	scan func(*sql.Rows) (T, error)
+	rows    *sql.Rows
+	scan    func(*sql.Rows) (T, error)
+	onClose func()
 }
 
 func (iter *tableIterator[T]) Next() bool { return iter.rows.Next() }
-func (iter *tableIterator[T]) Close()     { iter.rows.Close() }
+func (iter *tableIterator[T]) Close() {
+	iter.rows.Close()
+	if iter.onClose != nil {
+		iter.onClose()
+	}
+}
 func (iter *tableIterator[T]) GetItem() (T, error) {
 	return iter.scan(iter.rows)
 }
 
 // MakeOnlineAccountsIter creates an onlineAccounts iterator.
-func MakeOnlineAccountsIter(ctx context.Context, q db.Queryable, useStaging bool) (trackerdb.TableIterator[*encoded.OnlineAccountRecordV6], error) {
+func MakeOnlineAccountsIter(ctx context.Context, q db.Queryable, useStaging bool, excludeBefore basics.Round) (trackerdb.TableIterator[*encoded.OnlineAccountRecordV6], error) {
 	table := "onlineaccounts"
 	if useStaging {
 		table = "catchpointonlineaccounts"
 	}
 
+	var onClose func()
+	if excludeBefore != 0 {
+		// cheat: use Rdb to make a temporary table that we will delete later
```

Comment:
I suggest to explain in greater details why do you create a temp table. Maybe copying part of this PR description, imo such kind of doc must live in the codebase

Reply:
+1 a brief description of what's going on here would be great